app/graphql/mutations/system/associate_profiles.rb

```diff
@@ -11,26 +11,14 @@ class AssociateProfiles < BaseMutation
       field :system, ::Types::System, null: true
 
       def resolve(args = {})
-        host = find_host(args[:id])
+        host = find_hosts([args[:id]]).first
```

Comment:
Isn't this going to return an empty scope when the `id` passed is one for a new host? Even if that was not the case (e.g in `Mutations::Profile::AssociateSystems`), this would return always already existing host and create hosts without profile associations in the db. 

```
      existing_systems = ::Pundit.policy_scope(current_user, ::Host).where(id: system_ids)
      save_hosts(system_ids - existing_systems.pluck(:id))
      existing_systems
```

The last line should return existing_systems + systems created in `save_hosts`

Reply:
The last line does return existing_systems + systems created in save_hosts. As an AR::Relation, it doesn't resolve (make a DB query) until it's used. If save_hosts was successful, existing_systems now includes those new hosts.

Reply:
I've updated it to return `existing_systems.reload` for clarity. In testing this, if you call `existing_systems` without the pluck, then the response gets cached. If you don't do that, it works fine:

```rb
From: /app/app/graphql/mutations/base_mutation.rb @ line 35 Mutations::HostHelper#find_hosts:

    31: def find_hosts(system_ids)
    32:   existing_systems = ::Pundit.policy_scope(current_user, ::Host)
    33:                              .where(id: system_ids)
    34:   require 'pry'; binding.pry
 => 35:   save_hosts(system_ids - existing_systems.pluck(:id))
    36:   require 'pry'; binding.pry
    37:   existing_systems
    38: end

[1] pry(#<Mutations::System::AssociateProfiles>)> existing_systems.pluck(:id)
   (1.0ms)  SELECT "hosts"."id" FROM "hosts" WHERE "hosts"."account_id" = $1 AND "hosts"."id" = $2  [["account_id", "de350b90-63bd-47cb-aefe-76906b97ec84"], ["id", "4b34f310-b062-4680-8d75-d006b962db6c"]]
  ↳ (pry):3
=> []
[2] pry(#<Mutations::System::AssociateProfiles>)> exit
  CACHE  (0.0ms)  SELECT "hosts"."id" FROM "hosts" WHERE "hosts"."account_id" = $1 AND "hosts"."id" = $2  [["account_id", "de350b90-63bd-47cb-aefe-76906b97ec84"], ["id", "4b34f310-b062-4680-8d75-d006b962db6c"]]
  ↳ app/graphql/mutations/base_mutation.rb:35
  Account Load (0.6ms)  SELECT  "accounts".* FROM "accounts" WHERE "accounts"."id" = $1 LIMIT $2  [["id", "de350b90-63bd-47cb-aefe-76906b97ec84"], ["LIMIT", 1]]
  ↳ app/graphql/mutations/base_mutation.rb:19
  Host Load (1.0ms)  SELECT  "hosts".* FROM "hosts" WHERE "hosts"."id" = $1 AND "hosts"."account_id" = $2 LIMIT $3  [["id", "4b34f310-b062-4680-8d75-d006b962db6c"], ["account_id", "de350b90-63bd-47cb-aefe-76906b97ec84"], ["LIMIT", 1]]
  ↳ app/graphql/mutations/base_mutation.rb:48
   (0.8ms)  BEGIN
  ↳ app/graphql/mutations/base_mutation.rb:53
  Host Create (1.0ms)  INSERT INTO "hosts" ("id", "name", "created_at", "updated_at", "account_id") VALUES ($1, $2, $3, $4, $5) RETURNING "id"  [["id", "4b34f310-b062-4680-8d75-d006b962db6c"], ["name", "rhel7-insights-client.virbr0.akofink-laptop"], ["created_at", "2020-02-26 16:12:55.674838"], ["updated_at", "2020-02-26 16:12:55.674838"], ["account_id", "de350b90-63bd-47cb-aefe-
76906b97ec84"]]
  ↳ app/graphql/mutations/base_mutation.rb:53
   (4.5ms)  COMMIT
  ↳ app/graphql/mutations/base_mutation.rb:53

From: /app/app/graphql/mutations/base_mutation.rb @ line 37 Mutations::HostHelper#find_hosts:

    31: def find_hosts(system_ids)
    32:   existing_systems = ::Pundit.policy_scope(current_user, ::Host)
    33:                              .where(id: system_ids)
    34:   require 'pry'; binding.pry
    35:   save_hosts(system_ids - existing_systems.pluck(:id))
    36:   require 'pry'; binding.pry
 => 37:   existing_systems
    38: end

[1] pry(#<Mutations::System::AssociateProfiles>)> existing_systems.pluck(:id)
   (1.1ms)  SELECT "hosts"."id" FROM "hosts" WHERE "hosts"."account_id" = $1 AND "hosts"."id" = $2  [["account_id", "de350b90-63bd-47cb-aefe-76906b97ec84"], ["id", "4b34f310-b062-4680-8d75-d006b962db6c"]]
  ↳ (pry):4
=> ["4b34f310-b062-4680-8d75-d006b962db6c"]
[2] pry(#<Mutations::System::AssociateProfiles>)> existing_systems.first
  Host Load (1.1ms)  SELECT  "hosts".* FROM "hosts" WHERE "hosts"."account_id" = $1 AND "hosts"."id" = $2 ORDER BY "hosts"."id" ASC LIMIT $3  [["account_id", "de350b90-63bd-47cb-aefe-76906b97ec84"], ["id", "4b34f310-b062-4680-8d75-d006b962db6c"], ["LIMIT", 1]]
  ↳ (pry):5
=> #<Host:0x00007faa201d8f68 id: "4b34f310-b062-4680-8d75-d006b962db6c", name: "rhel7-insights-client.virbr0.akofink-laptop", created_at: Wed, 26 Feb 2020 16:12:55 UTC +00:00, updated_at: Wed, 26 Feb 2020 16:12:55 UTC +00:00, account_id: "de350b90-63bd-47cb-aefe-76906b97ec84">
[3] pry(#<Mutations::System::AssociateProfiles>)>
```